subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/TestFile.java

```diff
@@ -185,7 +185,7 @@ public TestFile setText(String content) {
     public TestFile[] listFiles() {
         File[] children = super.listFiles();
         if (children == null) {
-            return null;
+            return new TestFile[0];
```

Comment:
would NPE in the test otherwise since unreadable folders don't have content

Reply:
Why change this? In production code `File.listFiles()` will also return `null` for an unreadable file.

Reply:
this method was meant to customize `listFiles` anyway. You're saying this is too much?

Reply:
Reverted